### PR TITLE
[FLINK-32675][doc] Add document for the tiered storage of hybrid shuffle

### PR DESCRIPTION
**Pull request by TanYuxin-tyx**
## What is the purpose of the change

*Add document for the tiered storage of hybrid shuffle*

## Brief change log

  - *Add document for the tiered storage of hybrid shuffle*

## Verifying this change

This change is a doc update without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: docs/content.zh/docs/ops/batch/batch_shuffle.md [hunk 1]
```diff
@@ -127,6 +127,12 @@ These could be configured via [jobmanager.partition.hybrid.partition-data-consum
...
+
+### Remote Storage Support
+The **new Hybrid Shuffle mode** supports to store the shuffle data to the remote storage. The remote storage path can be configured by [taskmanager.network.hybrid-shuffle.remote.path]({{< ref "docs/deployment/config" >}}#taskmanager-network-hybrid-shuffle-remote-path). This feature supports various remote storage systems, including OSS, HDFS, S3, etc. See [Flink Filesystem]({{< ref "docs/deployment/filesystems/overview" >}}) for more information about the Flink supported filesystems.
```
[the remark this answers is not in the document]
Comment (TanYuxin-tyx) [reply docs/content.zh/docs/ops/batch/batch_shuffle.md]: Moved the section and added the note.

File: docs/content.zh/docs/ops/batch/batch_shuffle.md [hunk 2]
```diff
@@ -127,6 +127,12 @@ These could be configured via [jobmanager.partition.hybrid.partition-data-consum
...
 
+### The New Mode And The Legacy Mode
+A **new Hybrid Shuffle mode** supporting remote storage is introduced. The new mode has resolved existing issues in the legacy Hybrid Shuffle mode. Firstly, the new mode uses less required network memory. Secondly, the new mode can store shuffle data in remote storage when the disk space is not enough, which could avoid insufficient disk space errors. The new mode is only enabled when [taskmanager.network.hybrid-shuffle.remote.path]({{< ref "docs/deployment/config" >}}#taskmanager-network-hybrid-shuffle-remote-path) is configured. Currently, the new Hybrid Shuffle mode is in an experimental phase. In case of unexpected issues, the new mode can fall back to the legacy mode by setting [taskmanager.network.hybrid-shuffle.enable-new-mode]({{< ref "docs/deployment/config" >}}#taskmanager-network-hybrid-shuffle-enable-new-mode) to `false`. Once the new mode is stable, the legacy mode and the fallback option will be removed.
```
[the remark this answers is not in the document]
Comment (TanYuxin-tyx) [reply docs/content.zh/docs/ops/batch/batch_shuffle.md]: Fixed according to the comments.

**Comment by TanYuxin-tyx** [comment]
@xintongsong  Thanks for helping review. I have addressed the comments with a fix-up commit. PTAL.

**Review by xintongsong — APPROVED** [review]
LGTM. Merging.

**Pull request closed without merging**